rfc/20241206-oci-registries.md

```diff
@@ -0,0 +1,31 @@
+# OpenTofu Providers and Modules from OCI Registries
+
+Issue: https://github.com/opentofu/opentofu/issues/308
+
+## Background
+
+OCI registries (also historically known as Docker registries) form the backbone of the container ecosystem. They present an easy way to publish self-contained miniature operating system userland images that users can run without installing additional libraries or tools beyond a container engine. The main feature of an OCI registry is its ability to publish layers containing differential changes, making container image updates very efficient.
+
+However, thanks to the generic architecture of OCI registries, several implementations have popped up that allow users to store arbitrary data in OCI registries beyond container images. For example, [OCI Registry As Storage (ORAS)](https://oras.land/) is such a project, but the [Helm project also supports OCI storage](https://helm.sh/docs/topics/registries/).
```

Comment:
I don't suppose this really matters too much for what we're saying here, but FWIW I understood that Helm's use of OCI registries for distribution is an _application_ of ORAS, rather than something separate from it. Specifically, they seem to use the ORAS Go library as the foundation of their implementation.